src/include/platform/internal/GenericConfigurationManagerImpl.ipp

```diff
@@ -767,10 +841,13 @@ void GenericConfigurationManagerImpl<ConfigClass>::LogDeviceConfig()
 
     ChipLogProgress(DeviceLayer, "Device Configuration:");
 
+    DeviceInstanceInfoProvider * deviceInstanceInfoProvider = GetDeviceInstanceInfoProvider();
+
     {
         char serialNum[ConfigurationManager::kMaxSerialNumberLength + 1];
-        err = GetSerialNumber(serialNum, sizeof(serialNum));
-        ChipLogProgress(DeviceLayer, "  Serial Number: %s", (err == CHIP_NO_ERROR) ? serialNum : "(not set)");
+        MutableCharSpan serialNumber(serialNum);
+        err = deviceInstanceInfoProvider->GetSerialNumber(serialNumber);
+        ChipLogProgress(DeviceLayer, "  Serial Number: %s", (err == CHIP_NO_ERROR) ? serialNumber.data() : "(not set)");
```

Comment:
serialNumber may not be null-terminated. You may need to use "%.*s" formatting in this case.